Branch-SDK/src/io/branch/referral/SystemObserver.java

```diff
@@ -641,4 +644,31 @@ protected void onPostExecute(Void aVoid) {
     interface GAdsParamsFetchEvents {
         void onGAdsFetchFinished();
     }
+
+    /**
+     * Get IP address from first non local net Interface
+     */
+    public static String getLocalIPAddress() {
+        String ipAddress = "";
+        try {
+            List<NetworkInterface> netInterfaces = Collections.list(NetworkInterface.getNetworkInterfaces());
+            for (NetworkInterface netInterface : netInterfaces) {
+                List<InetAddress> addresses = Collections.list(netInterface.getInetAddresses());
+                for (InetAddress address : addresses) {
+                    if (!address.isLoopbackAddress()) {
+                        String ip = address.getHostAddress();
```

Comment:
Not only for that but I suspect a permission denial on some device with future android versions. As per adndroid policy ip is not allowed to access unless providing network permissions